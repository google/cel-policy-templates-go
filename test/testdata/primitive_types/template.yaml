# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

apiVersion: policy.acme.co/v1
kind: PolicyTemplate
metadata:
  name: primitive_types
  namespace: acme
schema:
  type: object
  properties:
    values:
      type: array
      items:
        type: object
        properties:
          bool:
            type: boolean
          float:
            type: number
          int:
            type: string
            format: int64
          uint:
            type: string
            format: uint64
          google_duration:
            type: string
            format: google-duration
            default: 300s
          google_timestamp:
            type: string
            format: google-datetime
            default: 1970-01-01T00:00:00Z
evaluator:
  productions:
    - decision: grant
      output: >
        string(rule.int + int(rule.uint)) + " " +
        string(duration("6s") + rule.google_duration) + " " +
        string(duration("86400s") + rule.google_timestamp))
