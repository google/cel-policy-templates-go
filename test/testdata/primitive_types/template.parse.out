1~2~apiVersion: 3~"policy.acme.co/v1"
4~kind: 5~"AdmissionTemplate"
6~metadata:7~
  8~name: 9~"AllTheTypes"
10~rules:11~
  - 12~13~value: 14~true
  - 15~16~value: 17~1.2
  - 18~19~value: 20~"1000"
  - 21~22~value: 23~
